packages/react-native-reanimated/src/valueSetter.ts

```diff
@@ -31,6 +31,7 @@ export function valueSetter<Value>(
     // built in animations that are not higher order(withTiming, withSpring) hold target value in .current
     if (
       mutable._value === animation.current &&
+      !animation.forceRunAnimation &&
```

Comment:
it seems like this check has a purpose (explained above), does introducing this flag not defeat that purpose? i think this needs to be rethought more holistically with the above comment in mind

Reply:
to be clear i'm not a maintainer, just thinking based on the code